### ISSUE TITLE
Branch 1.7.x build failed due to missing camel-bom transient dependency

### ISSUE THREAD

**Issue by alexkieling** [syndesisio/syndesis#6078]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
Branch 1.7.x build is failing due to the following error:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Integration 1.7-SNAPSHOT ........................... SUCCESS [  6.101 s]
[INFO] Integration :: Support :: Camel Acme ............... SUCCESS [ 15.613 s]
[INFO] Integration :: API ................................. SUCCESS [ 11.791 s]
[INFO] Integration :: Component Proxy ..................... SUCCESS [ 16.599 s]
[INFO] Integration :: Runtime ............................. SUCCESS [ 23.971 s]
[INFO] Integration :: Bill of Materials (BOM) ............. SUCCESS [  0.149 s]
[INFO] Integration :: Project Generator ................... SUCCESS [ 16.465 s]
[INFO] Integration :: Runtime :: Camel-k .................. FAILURE [ 11.290 s]
[INFO] Integration :: Camel K :: Bill of Materials (BOM) .. SKIPPED
[INFO] Integration :: Runtime :: SpringBoot 1.7-SNAPSHOT .. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:43 min
[INFO] Finished at: 2019-07-05T15:56:10-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin org.apache.camel.k:camel-k-maven-plugin:0.3.4.fuse-740007 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.camel.k:camel-k-maven-plugin:jar:0.3.4.fuse-740007: Could not find artifact org.apache.camel:camel-bom:pom:2.21.0.fuse-740037 in redhat-ga-repository (https://maven.repository.redhat.com/ga) -> [Help 1]
```

In the app/pom.xml, `camel-k-runtime.version` property is set to `0.3.4.fuse-740007`:
https://github.com/syndesisio/syndesis/blob/2a5af41ba0cd6639e6456f8c5cdd71e6f2d260c8/app/pom.xml#L81

`camel-k-runtime-parent` version `0.3.4.fuse-740007` depends on Camel version `2.21.0.fuse-740037`:
https://repository.jboss.org/nexus/content/groups/ea/org/apache/camel/k/camel-k-runtime-parent/0.3.4.fuse-740007/camel-k-runtime-parent-0.3.4.fuse-740007.pom

Camel version `2.21.0.fuse-740037` has been purged in the Maven repository:
https://repository.jboss.org/nexus/content/groups/ea/org/apache/camel/camel-bom/2.21.0.fuse-740037/

The only valid version of `camel-k-runtime-parent` seems to be `0.3.4.fuse-740001`, which references Camel version `2.21.0.fuse-740028`.

**Comment by alexkieling** [comment]
@nicolaferraro What do you think?

**Comment by nicolaferraro** [comment]
@alexkieling yes, I've encountered the same issue.

I think this won't affect the release pipeline, since everything is built from source there. So the best thing to do is to run a dev build of 7.4 and then pin both camel and camel k runtime to avoid another purge. After that we can switch to those versions on the syndesis 1.7.x branch.

The problem is that currently the release pipeline skips the runtime repo:
```
|= Project: camel v. 2.21.0.fuse-740037
Git Repo: git@github.com:jboss-fuse/camel.git
Git Tag: camel-2.21.0.fuse-740037
Build Skipped!

...


|= Project: camel-k-runtime v. 0.3.4.fuse-740007
Git Repo: git@github.com:jboss-fuse/camel-k-runtime.git
Git Tag: camel-k-runtime-0.3.4.fuse-740007
Build Skipped!
```

So we don't have a new build we can pin.

@alexkieling , @jamesnetherton, @cunningt do you know how to disable repo skipping? (probably doing an empty commit would help, but I'm not sure how it works).

**Comment by jamesnetherton** [comment]
I think you can set `skipOverride = false` on the `buildProject` closure to force project rebuilds. Else an empty commit would probably work.

**Comment by nicolaferraro** [comment]
Fixed by #6074 on branch 1.7.
